torch/utils/mkldnn.py

```diff
@@ -131,6 +131,153 @@ def forward(self, x):
             False,  # cuda_enabled
         )
 
+MKLDNN_RNN_RELU = 0
+MKLDNN_RNN_TANH = 1
+MKLDNN_LSTM = 2
+MKLDNN_GRU = 3
+
+class MkldnnLSTM(torch.jit.ScriptModule):
+    __constants__ = ['input_size', 'hidden_size', 'bias',
+                     'mode', 'num_layers', 'bidirectional', 'batch_first']
+
+    def __init__(self, dense_module):
+        super(MkldnnLSTM, self).__init__()
+
+        assert(not dense_module.training)
+        assert(not dense_module.batch_first)
+
+        self.input_size = dense_module.input_size
+        self.hidden_size = dense_module.hidden_size
+        self.bias = dense_module.bias
+        self.mode = MKLDNN_LSTM
+        self.num_layers = dense_module.num_layers
+        self.bidirectional = dense_module.bidirectional
+        self.batch_first = dense_module.batch_first
+
+        self.register_buffer(
+            'flatten_weight',
+            torch._C._nn._mkldnn_rnn_flatten_weight(
+                dense_module._flat_weights,
```

Comment:
it's a bit dangerous to rely on the internal order. It might be safer to copy-paste flat_weight fetching code here as the names of the parameters are more guaranteed to be static (https://github.com/pytorch/pytorch/blob/master/torch/nn/modules/rnn.py#L63)